packages/@lwc/template-compiler/src/state.ts

```diff
@@ -27,7 +28,17 @@ export default class State {
      */
     crCheckedElements: Map<BaseElement, boolean>;
 
-    constructor(config: NormalizedConfig) {
+    /**
+     * Filename of the HTML source
+     */
+    filename: string;
+
+    /**
+     * Contains the scopeTokens used in the template metadata
+     */
+    scopeTokens: scopeTokens;
+
+    constructor(config: NormalizedConfig, filename: string) {
```

Comment:
I understand why the `filename` isn't part of the `config`, but it strikes me as a bit odd... might be worth a comment.

Reply:
Updated in the latest commit! - 85f8a18f3c189c3fed2f07f0b7233ddf8ab6f117